lib/core/src/Cardano/Api/Gen.hs

```diff
@@ -264,6 +264,9 @@ genEncodingBoundaryLovelace = do
     offset <- frequency
         [ (1, choose (-10, 10))
 
+        -- Either offset by -1 (just below boundary), or 0 (just above boundary)
+        , (1, choose (-1, 0))
```

Comment:
Question: what is the benefit of choosing `0` here?

Reply:
That we make sure to generate e.g. `2^32`, which is encoded as 9 bytes, not just `2^32 - 1` which is 5 bytes.